server/api/crud/runtimes/nuclio/function.py

```diff
@@ -493,6 +493,24 @@ def _resolve_and_set_build_requirements_and_commands(function, config):
     mlrun.utils.update_in(config, "spec.build.commands", commands)
 
 
+def _enrich_function_node_selector_with_project(
+    run_db, project_name, function_node_selector
+):
+    config_node_selector = mlrun.mlconf.get_default_function_node_selector()
+    project_node_selector = {}
+
+    if run_db and project_name:
+        if project := run_db.get_project(project_name):
+            project_node_selector = project.spec.default_function_node_selector
+
+    # If either project node selector or config node selector is defined, enrich the config node selector
+    if project_node_selector or config_node_selector:
+        return mlrun.utils.helpers.merge_dicts_with_precedence(
+            config_node_selector, project_node_selector, function_node_selector
+        )
+    return function_node_selector
```

Comment:
Is there a reason we can't use one of the multiple other versions of this function that already exist?

I know some of the implementations have some runtime-specific intricacies, but adding the same functionality in each runtime is not maintainable at all.
See how can you common-ize these functions as much as you can.